### ISSUE TITLE
Loss weights

### ISSUE THREAD

**Issue by acasamitjana** [ucl-candi/DeepReg#153]
# Issue description

The total loss is a weighted sum of several losses. The weight associated with the label loss is not used in the code.

## Type of Issue

Please delete options that are not relevant.

- [ ] Bug report

#### For bug report or feature request, steps to reproduce the issue, including (but not limited to):

Run a training demo with labels

#### What's the expected result?

label_loss = weighted_loss * label_weight

#### What's the actual result?

label_loss = weighted_loss

**Comment by mathpluscode** [comment]
ooops, i confirm it's a bug, fix coming in 30 mins

**Comment by mathpluscode** [comment]
https://github.com/ucl-candi/DeepReg/blob/master/deepreg/model/network/util.py#L197 
there are multiple problems:

1. the condition should also check if the weight is >0
2. the weighted loss need to use the weight

and some more improvements
1. in https://github.com/ucl-candi/DeepReg/blob/master/deepreg/config/parser.py#L77, if the data is labeled, the label loss must have positive weight, otherwise no meaning of the label? Maybe a warning instead of ValueError? as people might want to do comparison without changing too much the config
2. as we are modifying this, maybe add a check of the regularization loss in case of ddf/dvf mode, but this should be in another issue, i will raise it

**Comment by YipengHu** [comment]
I will approve #154 first. @mathpluscode can you raise a separate issue for these issues you mentioned?
positive weight - yes, but should be a warning (there might be weird usage for the negative value?)

**Comment by mathpluscode** [comment]
ok @YipengHu go ahead
